### PR TITLE
Migrate websub module to new service typing

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
$subject.

Some of the tests are commented out as they are yet to be supported by other modules.

### CODE REVIEW

**Comment by Bhashinee** [comment]
Some of the tests are commented out as they are yet to be supported by other modules.

File: websub-generic-integration-tests/tests/test_custom_subscribers.bal [hunk 1]
```diff
@@ -32,88 +30,80 @@ public type CustomSubMockDomainEvent record {|
     string domain;
 |};
 
-@websub:SubscriberServiceConfig {
-    path:"/key"
-}
-service customSubKeyWebhook on new CustomSubWebhookServerForPayload(23585) {
-    resource function onIntentVerification(websub:Caller caller, websub:IntentVerificationRequest verRequest) {
+@websub:SubscriberServiceConfig {}
```
Comment (chamil321) [inline websub-generic-integration-tests/tests/test_custom_subscribers.bal:33]: Do we need this `websub:SubscriberServiceConfig` anymore as it's empty?

File: websub-generic-integration-tests/tests/test_subscriber_startup.bal [hunk 2]
```diff
@@ -24,37 +24,37 @@ import ballerina/websub;
...
-    resource function 'start(http:Caller caller, http:Request request) returns error? {
+service websub:SubscriberService /subscriber on testSubscriber {
+    resource function get 'start(http:Caller caller, http:Request request) returns error? {
 
         websub:Listener l1 = new(23387);
         websub:Listener l2 = new(23387);
 
         string responseMsg = "";
-        var l1Error = l1.__start();
+        var l1Error = l1.start();
```
Comment (chamil321) [inline websub-generic-integration-tests/tests/test_subscriber_startup.bal:34]:
```suggestion
        var l1Error = l1.'start();
```

File: websub-native/src/main/java/org/ballerinalang/net/websub/nativeimpl/SubscriberNativeOperationHandler.java [hunk 3]
```diff
@@ -224,12 +224,14 @@ private static void populateResourceDetails(HashMap<String, RecordType> resource
...
      */
-    public static void registerWebSubSubscriberService(BObject subscriberServiceListener, BObject service) {
+    public static void registerWebSubSubscriberService(BObject subscriberServiceListener, BObject service,
+            Object serviceName) {
         BObject serviceEndpoint = (BObject) subscriberServiceListener.get(
                 StringUtils.fromString(LISTENER_SERVICE_ENDPOINT));
         WebSubServicesRegistry webSubServicesRegistry =
                 (WebSubServicesRegistry) serviceEndpoint.getNativeData(WEBSUB_SERVICE_REGISTRY);
-        webSubServicesRegistry.registerWebSubSubscriberService(service);
+        String[] basePathArr = ((BArray) serviceName).getStringArray();
```
Comment (chamil321) [inline websub-native/src/main/java/org/ballerinalang/net/websub/nativeimpl/SubscriberNativeOperationHandler.java:233]: Need to handle this basePath for type string|string[]|() right?

**Review by niveathika — APPROVED** [review]

File: websub-generic-integration-tests/tests/test_custom_subscribers.bal (hunk 1 above)
Comment (Bhashinee) [reply websub-generic-integration-tests/tests/test_custom_subscribers.bal:33]: This becomes empty only for these custom subscribers. If we are to use Ballerina SubscriberService which is given by the websub module, this is compulsory to have with certain mandatory config fields. So there is a validation done to check whether this is present.

Comment (chamil321) [reply websub-generic-integration-tests/tests/test_custom_subscribers.bal:33]: Ack

**Review by chamil321 — APPROVED** [review]

**Pull request merged**